src/Http/Http.Features/src/FeatureReferences.cs

```diff
@@ -108,5 +108,17 @@ public void Initalize(IFeatureCollection collection, int revision)
 
         public TFeature Fetch<TFeature>(ref TFeature cached, Func<IFeatureCollection, TFeature> factory)
             where TFeature : class => Fetch(ref cached, Collection, factory);
+
+        private static int ContextDisposed()
+        {
+            ThrowContextDisposed();
+            return 0;
+        }
+
+        private static void ThrowContextDisposed()
```

Comment:
This part of the stack unfortunately can't assume the feature collection is HttpContext based.

Reply:
Changed